### DISCUSSION TITLE
how to properly load stimulus into Cognition.run?

### DISCUSSION THREAD

**Discussion by sonhkim** [jspsych/jsPsych#1637] [Q&A] [answered]
Hi all! 🙂

My jsPsych experiment runs without an issue on its own, but I can't make it run on Cognition.run. I think it has to do with the stimulus file ("stims.js"). 

So here's what I am doing: I upload the stims.js file using the window which is on the left side of Cognition.run. Then, I'm unclear what I should do the next. In jsPsych, you can do <script src = "path to stim file">. But how do you do that with Cognition.run? Does Cognition automatically recognize the variable "stims" that's defined in the stim.js file? (It seems not). 

So far, I tried the following line in the code: 
var stims = "stims.js";

but this doesn't work (please see the first screenshot). When I go into the console, it says the following: 
jspsych.js:965 You must specify a value for the prompt parameter (nested in the questions parameter) in the survey-likert plugin.

If I just define the variable "stims" _within_ the code (i.e., uncomment lines 16-19), then the problem goes away and the experiment starts running on Cognition.run. So what might be the problem? 

The stims.js file has the following content in it: 

var stims = [
{"attribute": "xx", "word": "yy", "question": "blablah"},
 {"attribute": "xx", "word": "yy", "question": "blablah"}.
]

![Screen Shot 2021-03-15 at 10 01 40 PM](https://user-images.githubusercontent.com/39136886/111249764-36948580-85da-11eb-8af3-262f1335a882.png)

Thank you for your help in advance!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @sonhkim, yes if you upload your stims.js file in cognition.run then the `stims` variable that's defined in that file should be available. You shouldn't have to do anything else. Can you try removing this line:
```js
var stims = "stims.js";
```
Because that will overwrite the `stims` variable that's defined in your file.

I'd that doesn't fix it then I'm not sure what else could be causing the error. Maybe double-check that there aren't any errors in your stims.js file. And you can check that you have access to it in the main code by adding `console.log(stims)` somewhere. Then if you preview the study with the developer tools open, you should see the stims array in the console.

**Reply by sonhkim** [reply]
Hi @becky-gilbert, thank you for such a prompt reply! I tried that but it didn't work. The console now directly complained about stims not being defined: code.js:13 Uncaught ReferenceError: stims is not defined. I don't know why this doesn't work for me.

But guess what, I uploaded the stim.js file into _the "external JS/CSS" window_, not the stims window, and removed the following line from the code, as you suggested, 
var stims = "stims.js";
and now it works like a charm!!! 🥳

Much thanks to the following thread (https://github.com/jspsych/jsPsych/discussions/1227#discussioncomment-138758).
